### ISSUE TITLE
When data change the pointerConfig feature is not updating

### ISSUE THREAD

**Issue by giltheodor** [Abhinandan-Kushwaha/react-native-gifted-charts#456]
Thank you for the great package.

Describe the bug
I have a screen where I am showing data in weekly and monthly base. I am using the pointerConfig feature with the prop persistPointer={true} .
When I change period the strip remain at the previous index and is not updating or return to the initial index (initialPointerIndex={0}).

Below a sample of he bug:

https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/111180498/61804b53-fa20-40bf-b073-9daf480a75c5

**Comment by giltheodor** [comment]
Is there any update for this bug?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @giltheodor 👋
Thanks for pointing out this bug. It has now been fixed in version `1.3.25`

I have also added a prop named `resetPointerOnDataChange` whose default value is `true`. Set it to `false` if you want the pointer to remain at its position after data change.

**Comment by giltheodor** [comment]
Thank you for your response. I used the `resetPointerOnDataChange` prop as you suggested. 

Also I have a container where I want to show the current values of both lines, for this reason I am using a useState where I pass a value in order to filter my raw data. 

As you can observe in the sample below the `pointerConfig` with the use of the useState and the `resetPointerOnDataChange` prop move the strip to the initial position.

https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/111180498/6ac3f165-d69f-4101-a875-5980e1b8dc4b

If I remove the useState or the `resetPointerOnDataChange` prop the strip stays in the selected index.

pointerLabelComponent: items => {
                setSelectedItem(items[0].day);
                return <LabelComponent items={items} />;
              },

Is there another way to achieve this task?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @giltheodor 
The default value of `resetPointerOnDataChange` is `true`, so you don't need to use it. Use it if you want to set it to false.

> Is there another way to achieve this task?

Do you want to achieve this without using the `useState`? If yes, how?

**Comment by giltheodor** [comment]
Hi,

Sorry for any inconvenience I may cost but it seems that I was clear in my comment above. 

I am not looking for another way to implement this but I was describing a bug.

The pointerConfig (strip, pointer and label component) always resets to the initial position when I stop pressing the line chart. This something that was not happening before we added the resetPointerOnDataChange prop.

Also, I observed that If I remove the useState that I send you above it seems that the pointerConfig is working as expected.

Thank you in advance
